app/src/main/java/com/example/myapplication/Constants.java

```diff
@@ -0,0 +1,15 @@
+package com.example.myapplication;
+
+import android.content.res.Resources;
+
+public class Constants {
```

Comment:
ok.. but once you have string recsources, you can use them on java classes directly and elegantly.
example from our previous project: 
![image](https://user-images.githubusercontent.com/49907259/72318414-e0117b80-36a4-11ea-8f11-675898936a0c.png)

another useful feature of string resources is **placeholders**.
example (from our previous project :smile_cat: ):
![image](https://user-images.githubusercontent.com/49907259/72318598-5e6e1d80-36a5-11ea-9c4b-24743ed04157.png)

![image](https://user-images.githubusercontent.com/49907259/72318249-6e393200-36a4-11ea-9844-3d58eee1f067.png)
('username' was a string variable, note the extra parameter to 'getString')

Reply:
I agree.

Reply:
but notice that if you are not inside an activity class, you will have to use Resources.getSystem.getString() ... a bit less elegant